src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -199,6 +199,19 @@ private BoundExpression BindArgListOperator(InvocationExpressionSyntax node, Dia
                 {
                     Error(diagnostics, ErrorCode.ERR_CantUseVoidInArglist, argument.Syntax);
                 }
+
+                switch (analyzedArguments.RefKind(i))
+                {
+                    case RefKind.None:
+                    case RefKind.Ref:
+                        break;
+                    case RefKind.In:
+                    case RefKind.Out:
+                        Error(diagnostics, ErrorCode.ERR_CantUseInOrOutInArglist, argument.Syntax);
+                        break;
+                    default:
+                        throw ExceptionUtilities.UnexpectedValue(analyzedArguments.RefKind(i));
```

Comment:
>throw ExceptionUtilities.UnexpectedValue(analyzedArguments.RefKind(i)); [](start = 24, length = 71)

I think it is better to not throw here, but instead check for kinds that we support. We support ```None``` and ```Ref```, for everything else it is safe to report an error and I believe it is better than throwing.

Reply:
What do you suggest to put in the diagnostic message instead? the notion of "refkind" is not public or intuitive, and the current one only mentions `in` and `out`.

Reply:
> What do you suggest to put in the diagnostic message instead? the notion of "refkind" is not public or intuitive, and the current one only mentions in and out.

I suggest to go with the message you have. It covers what we have today and, once we add a new kind, we either adjust the condition of the message.